tests/migrations/test_optimizer.py

```diff
@@ -1159,6 +1159,43 @@ def test_rename_index(self):
             ]
         )
 
+    def test_add_rename_index(self):
```

Comment:
Tests may require some of those "does not optimize" checks to prove that unrelated indexes are not messed with… I think you'll find it to start failing as it reveals the issue I highlighted above.